test/unit/math/prim/fun/lub_transform_test.cpp

```diff
@@ -182,7 +180,72 @@ TEST(prob_transform, lub_constrain_matrix) {
       EXPECT_FLOAT_EQ(result(i), stan::math::lub_constrain(x(i), lb(i), ub(i)));
     }
     auto x_free = stan::math::lub_free(result, lb, ub);
-    // EXPECT_MATRIX_EQ(x, x_free);
+    for (size_t i = 0; i < x.size(); ++i) {
+      EXPECT_FLOAT_EQ(x.coeff(i), x_free.coeff(i));
+    }
+    EXPECT_THROW(stan::math::lub_constrain(x, lb, ub_bad),
+                 std::invalid_argument);
+  }
+  // matrix, real, real, lp
+  {
+    double lp0 = 0.0;
+    double lp1 = 0.0;
+    Eigen::VectorXd result = stan::math::lub_constrain(x, lbd, ubd, lp0);
+    for (size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i),
+                      stan::math::lub_constrain(x(i), lbd, ubd, lp1));
+    }
+    EXPECT_FLOAT_EQ(lp0, lp1);
```

Comment:
I didn't see prim tests for lub vectorization behavior of lp so I added them here.